server/src/server.js

```diff
@@ -219,7 +219,17 @@ app.use(function(req, res, next) {
     req.accountId = authInfo.accountId;
   }
   req.cookies = cookies;
-  req.cookies._csrf = cookies.get("_csrf"); // csurf expects a property
+  // The cookies library doesn't detect duplicates; check manually
+  let rawCookies = req.get("cookie") || "";
+  let pairs = rawCookies.split(";");
+  let csrfTokens = pairs.filter(item => { return item.match(/__Host-csrf=/); });
+  if (csrfTokens.length > 1) {
+    let exc = new Error("Duplicate CSRF cookies");
+    exc.headerValue = rawCookies;
+    captureRavenException(exc);
+    simpleResponse(res, "Bad request", 400);
+  }
+  req.cookies._csrf = cookies.get("__Host-csrf"); // csurf expects a property
```

Comment:
I think csurf is right.

Confusingly enough we're using the csurf module which uses the csrf module.